cputest/unaligned_stores.cpp

```diff
@@ -0,0 +1,204 @@
+#include <algorithm>
+#include <cinttypes>
+
+#include <gctypes.h>
+#include <ogc/irq.h>
+#include <wiiuse/wpad.h>
+
+#include "common/BitUtils.h"
+#include "common/hwtests.h"
+
+// This test covers the behavior described in https://bugs.dolphin-emu.org/issues/12565
+
+constexpr size_t PAGE_SIZE = 4096;
+constexpr uintptr_t CACHED_MEMORY_FLAG = 0x40000000;
```

Comment:
Nitpick: this isn't really a flag, it "just happens" that nintendo chose to configure the default BAT mapping so that 0x8... is a cached mapping and 0xC... an uncached mapping

Reply:
Yes. I just couldn't think of a better name for it.

Reply:
I'd do it with CACHED_MEM1_BASE / UNCACHED_MEM1_BASE and do -cached +uncached maybe?

(And I suspect libogc has the right macros for this too :-) )